### PR TITLE
New Plugin: UnsuppressEmbeds

### PR DESCRIPTION
**Pull request by HypedDomi**
Preview:

### CODE REVIEW

File: src/plugins/UnsuppressEmbeds.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { addContextMenuPatch, NavContextMenuPatchCallback, removeContextMenuPatch } from "@api/ContextMenu";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { Menu, PermissionStore, RestAPI } from "@webpack/common";
```
Comment (aamiaa) [inline src/plugins/UnsuppressEmbeds.tsx:22]:
```suggestion
import { Menu, PermissionStore, ChannelStore, UserStore, RestAPI } from "@webpack/common";
```

File: src/plugins/UnsuppressEmbeds.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+const EMBED_SUPPRESSED = 1 << 2;
+const MANAGE_MESSAGES = 1n << 13n;
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => () => {
+    const { message: { embeds, flags } } = props;
+
+    const canManageMessages = !!(PermissionStore.getChannelPermissions({ id: props.channel.id }) & MANAGE_MESSAGES);
```
Comment (aamiaa) [inline src/plugins/UnsuppressEmbeds.tsx:30]:
```suggestion
    const canManageMessages = !!(PermissionStore.getChannelPermissions({ id: props.channel.id }) & MANAGE_MESSAGES);
    const isOwnDM = author.id === UserStore.getCurrentUser().id && (channel.isDM() || channel.isGroupDM());
```

File: src/plugins/UnsuppressEmbeds.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    const isEmbedSuppressed = !!(flags & EMBED_SUPPRESSED);
+
+    if (!canManageMessages) return;
```
Comment (aamiaa) [inline src/plugins/UnsuppressEmbeds.tsx:33]:
```suggestion
    if (!canManageMessages && !isOwnDM) return;
```

**Review by aamiaa — COMMENTED** [review]
It would be nice if this worked on your own messages in dms as well

File: src/plugins/UnsuppressEmbeds.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import { addContextMenuPatch, NavContextMenuPatchCallback, removeContextMenuPatch } from "@api/ContextMenu";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { Menu, PermissionStore, RestAPI } from "@webpack/common";
+
+const EMBED_SUPPRESSED = 1 << 2;
+const MANAGE_MESSAGES = 1n << 13n;
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => () => {
+    const { message: { embeds, flags } } = props;
```
Comment (aamiaa) [inline src/plugins/UnsuppressEmbeds.tsx:28]: And this (forgot, sorry)
```suggestion
    const { message: { embeds, flags, author } } = props;
    const channel = ChannelStore.getChannel(props.channel.id);
```

**Comment by HypedDomi** [comment]
Completely forgot about DMs 😅

File: src/plugins/UnsuppressEmbeds.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => () => {
+    const { message: { author, embeds, flags } } = props;
+
+    const canManageMessages = !!(PermissionStore.getChannelPermissions({ id: props.channel.id }) & MANAGE_MESSAGES);
+    const isOwnDM = author.id === UserStore.getCurrentUser().id && (props.channel.isDM() || props.channel.isGroupDM());
+    if (!canManageMessages && !isOwnDM) return;
+
+    const isEmbedSuppressed = !!(flags & EMBED_SUPPRESSED);
+    const menuItem = (() => {
```
Comment (Vendicated) [inline src/plugins/UnsuppressEmbeds.tsx:35]: this is kinda weird, why not skip this entirely and call children.push() inline?

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Vendicated** [comment]
i think the current position is meh, it would be much better above delete message

Also would possibly make more sense being red? And finally adding an icon for proper consistency would be nice 
image

**Comment by HypedDomi** [comment]
It's now red for suppressing embeds and in the normal color for unsuppressing. I couldn't find a matching icon tho

**Comment by HypedDomi** [comment]
<details>
<summary>Unsuppressed Embed</summary>

image

</details>

<details>
<summary>Suppressed Embed</summary>

image

</details>

**Comment by Vendicated** [comment]
still needs an icon cause right now it looks reaaally out of place

**Comment by Vendicated** [comment]
how about this from the HideAttachments plugin? You can move it from HideAttachments to components/Icons

image

File: src/plugins/UnsuppressEmbeds.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+const EMBED_SUPPRESSED = 1 << 2;
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => {
+    const { message: { author, embeds, flags } } = props;
+
+    const isEmbedSuppressed = (flags & EMBED_SUPPRESSED) !== 0;
+    const canManageMessages = !!(PermissionStore.getChannelPermissions({ id: props.channel.id }) & PermissionsBits.MANAGE_MESSAGES);
+    const isOwnDM = author.id === UserStore.getCurrentUser().id && (props.channel.isDM() || props.channel.isGroupDM());
+    if (!canManageMessages && !isOwnDM) return;
```
Comment (Vendicated) [inline src/plugins/UnsuppressEmbeds.tsx:33]: couldn't these checks be removed? If the delete message button exists, you have perms

Comment (HypedDomi) [reply src/plugins/UnsuppressEmbeds.tsx:33]: The Button also exists for your own Messages when you don’t have perms

**Review by Vendicated — APPROVED** [review]

**Pull request merged**